api/s2n.h

```diff
@@ -253,9 +253,9 @@ extern int s2n_connection_get_server_protocol_version(struct s2n_connection *con
 extern int s2n_connection_get_actual_protocol_version(struct s2n_connection *conn);
 extern int s2n_connection_get_client_hello_version(struct s2n_connection *conn);
 extern int s2n_connection_client_cert_used(struct s2n_connection *conn);
-extern const char *s2n_connection_get_cipher(struct s2n_connection *conn);
+extern const char *s2n_connection_get_cipher_name(struct s2n_connection *conn);
 extern int s2n_connection_is_valid_for_cipher_preferences(struct s2n_connection *conn, const char *version);
-extern const char *s2n_connection_get_curve(struct s2n_connection *conn);
+extern const char *s2n_connection_get_curve_name(struct s2n_connection *conn);
```

Comment:
In the interest of starting over, It might be a good idea choose a better name than `curve` for this. I wasn't thinking generally enough when I originally added it. For one, this function does not make it clear whether it is the curve used for ECDHE key exchange or the curve used for signing in ECDSA. It is the curve used for **key exchange**. In addition, it is only concerned with ECDH key exchange, leaving no way to retrieve information from other key exchange types(FFDH, for example).

Maybe `s2n_connection_get_key_exchange_name` properly captures the goal here? It would require us to update the semantics to not just return the ec curve name, but also other key exchange types.